mbedtls/src/lib.rs

```diff
@@ -37,8 +35,11 @@ pub mod hash;
 pub mod pk;
 pub mod rng;
 pub use mbedtls_platform_support::self_test as self_test;
+#[cfg(feature = "ssl")]
 pub mod ssl;
+#[cfg(feature = "x509")]
 pub mod x509;
+#[cfg(any(feature = "x509", feature = "ssl"))]
```

Comment:
Is it fine to restrict `alloc` under x509 or ssl ? Because this is originally a public API interface.
I guess it's better to put it under its own feature and make it as dependency for x509 or ssl.

Reply:
Do you see this module being used on its own (i.e., even when `x509` and `ssl` features are not enabled)? If so, I like your suggestion, otherwise wouldn't it just add to an already long list of features that are hard to keep track of?

Reply:
For example, `pkcs12` also depends on `alloc`.

Reply:
My idea is more about if external user wants to use `alloc` module, they can turn it on without turn on `x509` or `ssl` feature.
But if you stronly like current approach, we could follow it. But you need to update it to also turn on with `pkcs12`

Reply:
> My idea is more about if external user wants to use alloc module, they can turn it on without turn on x509 or ssl feature.
I was thinking that this is not likely. If it's a real use-case, then adding an `alloc` feature makes sense. Since you are more familiar with the library, you should decide which approach is better.

Reply:
Ok, got it.
Let just keep it simple